### PR TITLE
feat: staggered rendering of sites on map

### PR DESCRIPTION
**Pull request by spaasis**
Currently our site rendering works as follows:
1. get site Ids matching our query
2. split the ids into batches of 200
3. loop through the batches to get all results
4. return all sites at once

This PR changes steps 3 and 4 so that after each batch is returned separately (using a generator function). This means that we start rendering and populating the map as soon as the first batch of 200 sites is received, improving UI responsivity.

# Testing
1. choose secchi depth
2. set time range from the beginning of 2023
3. fetch sites - should render the map quicker and update the sites as long as there are results to fetch

### CODE REVIEW

File: src/components/siteSelection/SiteMap.vue [hunk 1]
```diff
@@ -77,9 +78,9 @@ export default defineComponent({
     const currentHoverFeature = ref(null as IHoverData | null)
     const selectedFeatures = ref(new Collection())
 
-    const availableFeatures = searchParameterStore.availableSites.map((s) =>
-      s.createFeature()
-    )
+    const availableFeatures = computed(() => {
```
Comment (spaasis) [inline src/components/siteSelection/SiteMap.vue:81]: Make computed to make the property reactive to changes in the store

File: src/components/siteSelection/SiteSelection.vue [hunk 2]
```diff
@@ -19,7 +19,7 @@
             :aria-label="$t('$sitePlaceholder')"
           >
             <option :value="0" disabled hidden>
-              {{ $t('$sitePlaceholder') }}
+              {{ $t('$sitePlaceholder') + ' (' + availableSites.length + ')' }}
```
Comment (spaasis) [inline src/components/siteSelection/SiteSelection.vue:22]: Current site count added as a debug helper, but this felt pretty nice

File: src/components/siteSelection/SiteSelection.vue [hunk 3]
```diff
@@ -85,7 +86,7 @@ export default defineComponent({
       useWaterQualityStore
     ),
     availableSites() {
-      return this.searchParameterStore.availableSites
+      return this.searchParameterStore.availableSites.sort((s1, s2) => alphabeticCompare(s1.name, s2.name))
```
Comment (spaasis) [inline src/components/siteSelection/SiteSelection.vue:89]: Performing the ordering after each batch concatenation was slow, so moved the ordering here, where it matters more and did not cause such issues

File: src/components/siteSelection/SiteSelection.vue (hunk 2 above)
Comment (Vanamo) [reply src/components/siteSelection/SiteSelection.vue:22]: I'm not sure if it's easy to understand what the number stands for

**Comment by Vanamo** [comment]
Works nice!

File: src/components/siteSelection/SiteSelection.vue (hunk 2 above)
Comment (Vanamo) [reply src/components/siteSelection/SiteSelection.vue:22]: I think it could be removed

**Pull request merged**